### DISCUSSION TITLE
Problem in installing OpenASIP tools

### DISCUSSION THREAD

**Discussion by Jedrek09** [cpc/tce#269] [Q&A]
Hello . I hope this message finds you well.  There are certain issues I wanted to talk about while installing OpenASIP Tools.

I installed C++ and python and made sure pywin32 was installed correctly. Then I installed virtual box and downloaded the linux iso file. Then I finally created the virtual machine. 
 However, I encountered challenges while installing the OpenASIP tools. After ensuring approximately 44 GB of space on my machine, I followed the instructions closely. Unfortunately, I discovered that the script for installing LLVM 15 was missing from the repository which I required, with scripts only available for LLVM versions 14, 16, and 17. I attempted a manual installation of LLVM 15 but encountered an error in the CMakeLists file.  Consequently, I proceeded with the installation of LLVM 16, as it is an updated version of LLVM 15 containing all of its features, which was successful, and I implemented the code

The primary obstacle arose where I encountered errors related to deprecated Boost headers (progress.hpp and timer.hpp) and deprecated C++ functionality. To address these issues, I defined BOOST_TIMER_ENABLE_DEPRECATED in the build environment and made efforts to refactor the code to remove deprecated C++ features.  I resolved these by manually replacing deprecated headers with their modern equivalents and installing missing C++ headers that were not available by default on my system.

Unfortunately, I then encountered another issue where the compiler was unable to locate the MCInstrDesc.h header file from the LLVM project. I resolved this by manually identifying the correct path within the LLVM build and including it in the installation process. While this addressed the immediate problem, the Boost header issues reappeared.

Could you please provide any solution for that.

**Comment by Jedrek09** [comment]
I installed LLVM 16 by following the commands that were provided ad was successful in running it. I made sure there were no residual of llvm 15 or any other version. When running the final step that it ./autogen.sh && ./configure --prefix=$HOME/local && make -j8 && make install, I encountered the issue (OperationPoolPimpl.cc:60:10: fatal error: llvm/MC/MCInstrDesc.h: No such file or directory
   60 | #include <llvm/MC/MCInstrDesc.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[4]: *** [Makefile:575: OperationPoolPimpl.lo] Error 1
make[4]: *** Waiting for unfinished jobs....
mv -f .deps/OperationSerializer.Tpo .deps/OperationSerializer.Plo
mv -f .deps/OperationGlobals.Tpo .deps/OperationGlobals.Plo
mv -f .deps/OperationIndex.Tpo .deps/OperationIndex.Plo
mv -f .deps/OperationDAGBehavior.Tpo .deps/OperationDAGBehavior.Plo
mv -f .deps/OperationDAGBuilder.Tpo .deps/OperationDAGBuilder.Plo
mv -f .deps/OperationDAG.Tpo .deps/OperationDAG.Plo
mv -f .deps/OperationDAGConverter.Tpo .deps/OperationDAGConverter.Plo
make[4]: Leaving directory '/home/salman/openasip-devel/openasip/src/base/osal'
make[3]: *** [Makefile:467: all-recursive] Error 1
make[3]: Leaving directory '/home/salman/openasip-devel/openasip/src/base'
make[2]: *** [Makefile:575: all-recursive] Error 1
make[2]: Leaving directory '/home/salman/openasip-devel/openasip/src'
make[1]: *** [Makefile:567: all-recursive] Error 1
make[1]: Leaving directory '/home/salman/openasip-devel/openasip'
make: *** [Makefile:441: all] Error 2)

I again tried with ./autogen.sh && ./configure --prefix=$HOME/local && make -j1 && make install, and still got the same issue 

After running (dpkg-query -l | grep libboost)

I got the following
(ii  libboost-all-dev                              1.83.0.1ubuntu2                          amd64        Boost C++ Libraries development files (ALL) (default version)
ii  libboost-atomic-dev:amd64                     1.83.0.1ubuntu2                          amd64        atomic data types, operations, and memory ordering constraints (default version)
ii  libboost-atomic1.83-dev:amd64                 1.83.0-2.1ubuntu3                        amd64        atomic data types, operations, and memory ordering constraints
ii  libboost-atomic1.83.0:amd64                   1.83.0-2.1ubuntu3                        amd64        atomic data types, operations, and memory ordering constraints
ii  libboost-chrono-dev:amd64                     1.83.0.1ubuntu2                          amd64        C++ representation of time duration, time point, and clocks (default version)
ii  libboost-chrono1.83-dev:amd64                 1.83.0-2.1ubuntu3                        amd64        C++ representation of time duration, time point, and clocks
ii  libboost-chrono1.83.0t64:amd64                1.83.0-2.1ubuntu3                        amd64        C++ representation of time duration, time point, and clocks
ii  libboost-container-dev:amd64                  1.83.0.1ubuntu2                          amd64        C++ library that implements several well-known containers - dev files (default version)
ii  libboost-container1.83-dev:amd64              1.83.0-2.1ubuntu3                        amd64        C++ library that implements several well-known containers - dev files
ii  libboost-container1.83.0:amd64                1.83.0-2.1ubuntu3                        amd64        C++ library that implements several well-known containers
ii  libboost-context-dev:amd64                    1.83.0.1ubuntu2                          amd64        provides a sort of cooperative multitasking on a single thread (default version)
ii  libboost-context1.83-dev:amd64                1.83.0-2.1ubuntu3                        amd64        provides a sort of cooperative multitasking on a single thread
ii  libboost-context1.83.0:amd64                  1.83.0-2.1ubuntu3                        amd64        provides a sort of cooperative multitasking on a single thread
ii  libboost-coroutine-dev:amd64                  1.83.0.1ubuntu2                          amd64        provides a sort of cooperative multitasking on a single thread (default version)
ii  libboost-coroutine1.83-dev:amd64              1.83.0-2.1ubuntu3                        amd64        provides a sort of cooperative multitasking on a single thread
ii  libboost-coroutine1.83.0:amd64                1.83.0-2.1ubuntu3                        amd64        provides a sort of cooperative multitasking on a single thread
ii  libboost-date-time-dev:amd64                  1.83.0.1ubuntu2                          amd64        set of date-time libraries based on generic programming concepts (default version)
ii  libboost-date-time1.83-dev:amd64              1.83.0-2.1ubuntu3                        amd64        set of date-time libraries based on generic programming concepts
ii  libboost-date-time1.83.0:amd64                1.83.0-2.1ubuntu3                        amd64        set of date-time libraries based on generic programming concepts
ii  libboost-dev:amd64                            1.83.0.1ubuntu2                          amd64        Boost C++ Libraries development files (default version)
ii  libboost-exception-dev:amd64                  1.83.0.1ubuntu2                          amd64        library to help write exceptions and handlers (default version)
ii  libboost-exception1.83-dev:amd64              1.83.0-2.1ubuntu3                        amd64        library to help write exceptions and handlers
ii  libboost-fiber-dev:amd64                      1.83.0.1ubuntu2                          amd64        cooperatively-scheduled micro-/userland-threads (default version)
ii  libboost-fiber1.83-dev:amd64                  1.83.0-2.1ubuntu3                        amd64        cooperatively-scheduled micro-/userland-threads
ii  libboost-fiber1.83.0:amd64                    1.83.0-2.1ubuntu3                        amd64        cooperatively-scheduled micro-/userland-threads
ii  libboost-filesystem-dev:amd64                 1.83.0.1ubuntu2                          amd64        filesystem operations (portable paths, iteration over directories, etc) in C++ (default version)
ii  libboost-filesystem1.83-dev:amd64             1.83.0-2.1ubuntu3                        amd64        filesystem operations (portable paths, iteration over directories, etc) in C++
ii  libboost-filesystem1.83.0:amd64               1.83.0-2.1ubuntu3                        amd64        filesystem operations (portable paths, iteration over directories, etc) in C++
ii  libboost-graph-dev:amd64                      1.83.0.1ubuntu2                          amd64        generic graph components and algorithms in C++ (default version)
ii  libboost-graph-parallel-dev                   1.83.0.1ubuntu2                          amd64        generic graph components and algorithms in C++ (default version)
ii  libboost-graph-parallel1.83-dev               1.83.0-2.1ubuntu3                        amd64        generic graph components and algorithms in C++
ii  libboost-graph-parallel1.83.0                 1.83.0-2.1ubuntu3                        amd64        generic graph components and algorithms in C++
ii  libboost-graph1.83-dev:amd64                  1.83.0-2.1ubuntu3                        amd64        generic graph components and algorithms in C++
ii  libboost-graph1.83.0:amd64                    1.83.0-2.1ubuntu3                        amd64        generic graph components and algorithms in C++
ii  libboost-iostreams-dev:amd64                  1.83.0.1ubuntu2                          amd64        Boost.Iostreams Library development files (default version)
ii  libboost-iostreams1.83-dev:amd64              1.83.0-2.1ubuntu3                        amd64        Boost.Iostreams Library development files
ii  libboost-iostreams1.83.0:amd64                1.83.0-2.1ubuntu3                        amd64        Boost.Iostreams Library
ii  libboost-json-dev:amd64                       1.83.0.1ubuntu2                          amd64        C++ containers and algorithms that implement JSON (default version)
ii  libboost-json1.83-dev:amd64                   1.83.0-2.1ubuntu3                        amd64        C++ containers and algorithms that implement JSON
ii  libboost-json1.83.0:amd64                     1.83.0-2.1ubuntu3                        amd64        C++ containers and algorithms that implement JSON
ii  libboost-locale-dev:amd64                     1.83.0.1ubuntu2                          amd64        C++ facilities for localization (default version)
ii  libboost-locale1.83-dev:amd64                 1.83.0-2.1ubuntu3                        amd64        C++ facilities for localization
ii  libboost-locale1.83.0:amd64                   1.83.0-2.1ubuntu3                        amd64        C++ facilities for localization
ii  libboost-log-dev                              1.83.0.1ubuntu2                          amd64        C++ logging library (default version)
ii  libboost-log1.83-dev                          1.83.0-2.1ubuntu3                        amd64        C++ logging library
ii  libboost-log1.83.0                            1.83.0-2.1ubuntu3                        amd64        C++ logging library
ii  libboost-math-dev:amd64                       1.83.0.1ubuntu2                          amd64        Boost.Math Library development files (default version)
ii  libboost-math1.83-dev:amd64                   1.83.0-2.1ubuntu3                        amd64        Boost.Math Library development files
ii  libboost-math1.83.0:amd64                     1.83.0-2.1ubuntu3                        amd64        Boost.Math Library
ii  libboost-mpi-dev                              1.83.0.1ubuntu2                          amd64        C++ interface to the Message Passing Interface (MPI) (default version)
ii  libboost-mpi-python-dev                       1.83.0.1ubuntu2                          amd64        C++ interface to the Message Passing Interface (MPI), Python Bindings (default version)
ii  libboost-mpi-python1.83-dev                   1.83.0-2.1ubuntu3                        amd64        C++ interface to the Message Passing Interface (MPI), Python Bindings
ii  libboost-mpi-python1.83.0                     1.83.0-2.1ubuntu3                        amd64        C++ interface to the Message Passing Interface (MPI), Python Bindings
ii  libboost-mpi1.83-dev                          1.83.0-2.1ubuntu3                        amd64        C++ interface to the Message Passing Interface (MPI)
ii  libboost-mpi1.83.0                            1.83.0-2.1ubuntu3                        amd64        C++ interface to the Message Passing Interface (MPI)
ii  libboost-nowide-dev                           1.83.0.1ubuntu2                          amd64        Standard library functions with UTF-8 API on Windows development files (default version)
ii  libboost-nowide1.83-dev                       1.83.0-2.1ubuntu3                        amd64        Standard library functions with UTF-8 API on Windows development files
ii  libboost-nowide1.83.0                         1.83.0-2.1ubuntu3                        amd64        Standard library functions with UTF-8 API on Windows
ii  libboost-numpy-dev                            1.83.0.1ubuntu2                          amd64        Boost.Python NumPy extensions development files (default version)
ii  libboost-numpy1.83-dev                        1.83.0-2.1ubuntu3                        amd64        Boost.Python NumPy extensions development files
ii  libboost-numpy1.83.0                          1.83.0-2.1ubuntu3                        amd64        Boost.Python NumPy extensions
ii  libboost-program-options-dev:amd64            1.83.0.1ubuntu2                          amd64        program options library for C++ (default version)
ii  libboost-program-options1.83-dev:amd64        1.83.0-2.1ubuntu3                        amd64        program options library for C++
ii  libboost-program-options1.83.0:amd64          1.83.0-2.1ubuntu3                        amd64        program options library for C++
ii  libboost-python-dev                           1.83.0.1ubuntu2                          amd64        Boost.Python Library development files (default version)
ii  libboost-python1.83-dev                       1.83.0-2.1ubuntu3                        amd64        Boost.Python Library development files
ii  libboost-python1.83.0                         1.83.0-2.1ubuntu3                        amd64        Boost.Python Library
ii  libboost-random-dev:amd64                     1.83.0.1ubuntu2                          amd64        Boost Random Number Library (default version)
ii  libboost-random1.83-dev:amd64                 1.83.0-2.1ubuntu3                        amd64        Boost Random Number Library
ii  libboost-random1.83.0:amd64                   1.83.0-2.1ubuntu3                        amd64        Boost Random Number Library
ii  libboost-regex-dev:amd64                      1.83.0.1ubuntu2                          amd64        regular expression library for C++ (default version)
ii  libboost-regex1.83-dev:amd64                  1.83.0-2.1ubuntu3                        amd64        regular expression library for C++
ii  libboost-regex1.83.0:amd64                    1.83.0-2.1ubuntu3                        amd64        regular expression library for C++
ii  libboost-serialization-dev:amd64              1.83.0.1ubuntu2                          amd64        serialization library for C++ (default version)
ii  libboost-serialization1.83-dev:amd64          1.83.0-2.1ubuntu3                        amd64        serialization library for C++
ii  libboost-serialization1.83.0:amd64            1.83.0-2.1ubuntu3                        amd64        serialization library for C++
ii  libboost-stacktrace-dev:amd64                 1.83.0.1ubuntu2                          amd64        library to capture and print stack traces - development files (default version)
ii  libboost-stacktrace1.83-dev:amd64             1.83.0-2.1ubuntu3                        amd64        library to capture and print stack traces - development files
ii  libboost-stacktrace1.83.0:amd64               1.83.0-2.1ubuntu3                        amd64        library to capture and print stack traces
ii  libboost-system-dev:amd64                     1.83.0.1ubuntu2                          amd64        Operating system (e.g. diagnostics support) library (default version)
ii  libboost-system1.83-dev:amd64                 1.83.0-2.1ubuntu3                        amd64        Operating system (e.g. diagnostics support) library
ii  libboost-system1.83.0:amd64                   1.83.0-2.1ubuntu3                        amd64        Operating system (e.g. diagnostics support) library
ii  libboost-test-dev:amd64                       1.83.0.1ubuntu2                          amd64        components for writing and executing test suites (default version)
ii  libboost-test1.83-dev:amd64                   1.83.0-2.1ubuntu3                        amd64        components for writing and executing test suites
ii  libboost-test1.83.0:amd64                     1.83.0-2.1ubuntu3                        amd64        components for writing and executing test suites
ii  libboost-thread-dev:amd64                     1.83.0.1ubuntu2                          amd64        portable C++ multi-threading (default version)
ii  libboost-thread1.83-dev:amd64                 1.83.0-2.1ubuntu3                        amd64        portable C++ multi-threading
ii  libboost-thread1.83.0:amd64                   1.83.0-2.1ubuntu3                        amd64        portable C++ multi-threading
ii  libboost-timer-dev:amd64                      1.83.0.1ubuntu2                          amd64        C++ wall clock and CPU process timers (default version)
ii  libboost-timer1.83-dev:amd64                  1.83.0-2.1ubuntu3                        amd64        C++ wall clock and CPU process timers
ii  libboost-timer1.83.0:amd64                    1.83.0-2.1ubuntu3                        amd64        C++ wall clock and CPU process timers
ii  libboost-tools-dev                            1.83.0.1ubuntu2                          amd64        Boost C++ Libraries development tools (default version)
ii  libboost-type-erasure-dev:amd64               1.83.0.1ubuntu2                          amd64        C++ runtime polymorphism based on concepts (default version)
ii  libboost-type-erasure1.83-dev:amd64           1.83.0-2.1ubuntu3                        amd64        C++ runtime polymorphism based on concepts
ii  libboost-type-erasure1.83.0:amd64             1.83.0-2.1ubuntu3                        amd64        C++ runtime polymorphism based on concepts
ii  libboost-url-dev:amd64                        1.83.0.1ubuntu2                          amd64        C++ library that implements "URL" (default version)
ii  libboost-url1.83-dev:amd64                    1.83.0-2.1ubuntu3                        amd64        C++ library that implements "URL"
ii  libboost-url1.83.0:amd64                      1.83.0-2.1ubuntu3                        amd64        C++ library that implements "URL"
ii  libboost-wave-dev:amd64                       1.83.0.1ubuntu2                          amd64        C99/C++ preprocessor library (default version)
ii  libboost-wave1.83-dev:amd64                   1.83.0-2.1ubuntu3                        amd64        C99/C++ preprocessor library
ii  libboost-wave1.83.0:amd64                     1.83.0-2.1ubuntu3                        amd64        C99/C++ preprocessor library
ii  libboost1.83-dev:amd64                        1.83.0-2.1ubuntu3                        amd64        Boost C++ Libraries development files
ii  libboost1.83-tools-dev                        1.83.0-2.1ubuntu3                        amd64        Boost C++ Libraries development tools)

My Operating system is Ubuntu 24.04 LTS

**Comment by pjaaskel** [comment]
Converted the issue to discussion here: https://github.com/cpc/openasip/discussions/270?converting=1
